src/ansys/sound/core/sound_composer/source_spectrum.py

```diff
@@ -0,0 +1,238 @@
+            str_source_control = "Not set/valid"
+
+        return f"Spectrum source: {str_source}\nSource control: {str_source_control}"
+
+    @property
+    def source_control(self) -> SourceControlSpectrum:
+        """Spectrum source control object.
+
+        Contains the duration in seconds, and the method used to generate the sound.
+        """
+        return self.__source_control
+
+    @source_control.setter
+    def source_control(self, source_control: SourceControlSpectrum):
+        """Set the spectrum source control."""
+        if not (isinstance(source_control, SourceControlSpectrum) or source_control is None):
+            raise PyAnsysSoundException(
+                "Specified source control object must be of type ``SourceControlSpectrum``."
+            )
+        self.__source_control = source_control
+
+    @property
+    def source_spectrum_data(self) -> Field:
+        """Spectrum source data, as a DPF field.
+
+        Power spectral density (PSD) as a DPF field, which contains the frequencies in Hz and
+        the levels in unit^2/Hz (for example Pa^2/Hz).
+        """
+        return self.__source_spectrum_data
+
+    @source_spectrum_data.setter
+    def source_spectrum_data(self, source_spectrum_data: Field):
+        """Set the spectrum source data, from a DPF field."""
+        if not (isinstance(source_spectrum_data, Field) or source_spectrum_data is None):
+            raise PyAnsysSoundException(
+                "Specified spectrum source must be provided as a DPF field."
+            )
+        self.__source_spectrum_data = source_spectrum_data
+
+    def is_source_control_valid(self) -> bool:
+        """Source control verification function.
+
+        Check if the source control is set and its duration is strictly positive.
+
+        Returns
+        -------
+        bool
+            True if the source control is set and its duration is strictly positive, False
+            otherwise.
+        """
+        return (self.source_control is not None) and (self.source_control.duration > 0.0)
+
+    def load_source_spectrum(self, file_source: str):
+        """Load the spectrum source data from a spectrum file.
+
+        Parameters
+        ----------
+        file_source : str
+            Path to the spectrum file. Supported files are the same XML and text (with the
+            AnsysSound_Spectrum header) formats as supported by Ansys Sound SAS.
+        """
+        # Set operator inputs.
+        self.__operator_load.connect(0, file_source)
+
+        # Run the operator.
+        self.__operator_load.run()
+
+        # Get the loaded sound power level parameters.
+        self.source_spectrum_data = self.__operator_load.get_output(0, "field")
+
+    def process(self, sampling_frequency: float = 44100.0):
+        """Generate the sound of the spectrum source, using the current spectrum and source control.
+
+        Parameters
+        ----------
+        sampling_frequency : float, default 44100.0
```

Comment:
Add a check on sampling_frequency value + documentation: Fs > 0

Reply:
Done (except documentation, I feel it is unnecessary)